internal/generator/generator_util.go

```diff
@@ -184,12 +184,39 @@ func getMasksFromPath(path string) (directive2Masks map[string][]mask, err error
 	return directive2Masks, nil
 }
 
-func genFromSrcCode(codePath string, mapVariableName string, matchFnName string, writer io.Writer) error {
+func genFromSrcCode(codePath string, mapVariableName string, matchFnName string, writer io.Writer,
+	filter map[string]struct{}, override map[string][]mask) error {
 	directive2Masks, err := getMasksFromPath(codePath)
 	if err != nil {
 		return err
 	}
 
+	if filter != nil {
```

Comment:
recommend using `len(x) > 0` instead of `x != nil` when looking at maps and slices. `len` will return 0 for nils or non-nil but empty values, and we usually want to treat them the same.

```suggestion
	if len(filter) > 0 {
```